system_baseline/openapi/internal_api.spec.yaml

```diff
@@ -44,6 +44,37 @@ paths:
           $ref: '#/components/responses/Forbidden'
         '500':
           $ref: '#/components/responses/InternalServerError'
+  /systems/{system_id}:
+    parameters:
+      - $ref: '#/components/parameters/SystemId'
+    patch:
+      summary: update mapped system records
+      description: "update mapped system records"
```

Comment:
PATCH should be able to update fields. This is a very specific RPC call.
I'd suggest to create a subresource like: `PUT/PATCH /systems/{system_id}/groups`.

Alternatively to keep the same PATCH API under `/systems/{system_id}`, the request should patch `groups` field. Like `{groups: [{name: "Group Name", id: X}, ...}].

Reply:
You are right.

Currently we have only these fields on `system`:
`id, system_id, system_baseline_id, account, org_id, groups`.
Systems are "just" join table with some more metadata for querying but from API point of view we are considering them to be Systems (very very limited).
 
We are never going to change these `id, system_id (we query on it), system_baseline_id, account, org_id`. `system_baseline_id` is pointing to other record, `system_id` we can't change obviously as it points to real system record out of Drift - these two are "identity" of this record in baselines table. `account` and `org_id` are no going to be changed as these are set with the system object in Inventory.

This leaves only `groups` which can change in Inventory and we have them here as a key in `json` body of the request defined few lines below.

Unless I misunderstood your point, does that explain the situation? @vkrizan

Reply:
Yes, that was my point, to have a single key (`groups`) in the PATH request.

Reply:
I will update description of the endpoint that it accepts only `groups` at the time.